bcda/auth/models.go

```diff
@@ -63,3 +62,22 @@ func GetACOByClientID(clientID string) (models.ACO, error) {
 	}
 	return aco, err
 }
+
+// RevokeSystemKeyPair soft deletes the specified encryption key so that it can no longer be used
+func RevokeSystemKeyPair(encryptionKeyID uint) (models.EncryptionKey, error) {
+	db := database.GetGORMDbConnection()
+	defer database.Close(db)
+	encryptionKey := models.EncryptionKey{}
+
+	err := db.Find(&encryptionKey, encryptionKeyID).Error
+	if err != nil {
+		return encryptionKey, err
+	}
+
+	err = db.Delete(&encryptionKey).Error
```

Comment:
This isn't updating `encryptionKey`, so when it's returned, it doesn't look like it's deleted.  You _could_ add this just before returning:

````
	// Update encryptionKey's deleted_at value
	err = db.Unscoped().Find(&encryptionKey, encryptionKeyID).Error
	if err != nil {
		return encryptionKey, err
	}
````

Reply:
Or maybe it's better just to return `error`

Reply:
hmmm...  i'm going to return the deleted key like you showed above.  just in case we have an api or cli func that wants to return the deleted object

Reply:
Why would we return a deleted object? No one can use it for any practical purpose.

Reply:
yeah thats what i was originally doing.  honestly mostly did it for a higher fidelity unit test.  reverting now

Reply:
> a higher fidelity unit test

I assume by this you meant that you wanted to make sure the thing was deleted? You could something along the lines of the code below to the test:
```
err = db.Unscoped().Find(&deletedKey, encryptionKeyID).Error
assert.Nil(err)
assert.NotNil(deletedKey)
assert.NotNil(deletedKey.DeletedAt)
```